import/index_java/src/org/vufind/index/FormatCalculator.java

```diff
@@ -547,7 +547,7 @@ protected boolean hasSerialHost(Record record) {
 
     protected String getSubfieldOrDefault(DataField field, char subfieldCode, String defaultValue) {
         Subfield subfield = field.getSubfield(subfieldCode);
-        String data = subfield.getData();
+        String data = subfield != null ? subfield.getData() : null;
```

Comment:
Good catch! By the way, what happens if the subfield exists but is an empty string? Do we also want to return the defaultValue in this case? (I'm not sure if this is a proper case anyway)

Reply:
Empty subfields happen even if they shouldn't, and I'd return the default value in that case.

Reply:
I've added a simple empty subfield check. Is this what you had in mind? Note that I haven't had time to actually test it yet! I'll add a TODO to run the test suite before merging.

Reply:
Yes. I think `data.isEmpty()` would be the recommended way though, but I'd be happy to be corrected on this.

Reply:
I also think data.isEmpty() is the correct way to check for an empty string, else it might compare the pointer instead of the actual string value